### DISCUSSION TITLE
how about to rewrite client  using javafx?

### DISCUSSION THREAD

**Discussion by linxd** [nextgenhealthcare/connect#5616] [Feature/Enhancement Ideas]
is there sb think this?

**Comment by jonbartels** [comment]
I think a better idea would be to rewrite the client as a webapp instead of a local executable.

JavaFX is used in the MC Admin Launcher. It works pretty well but there have been some issues with it lately, especially for Mac users - https://github.com/nextgenhealthcare/connect/search?q=javafx&type=issues . This is not a problem with JavaFX but a problem with how the JVM packages JavaFX.

**Reply by siddharthsharma9258** [reply]
I think Rhapsody has a webapp?

**Comment by tonygermano** [comment]
A new client would require a plugin architecture and require that all existing client plugins (including third-party plugins) add support for this new architecture.
